autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/sync/CredentialsSyncMetadata.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.sync
+
+import com.duckduckgo.app.global.formatters.time.DatabaseDateFormatter
+import com.duckduckgo.autofill.store.CredentialsSyncMetadataDao
+import com.duckduckgo.autofill.store.CredentialsSyncMetadataEntity
+import com.duckduckgo.di.scopes.AppScope
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+@SingleInstanceIn(AppScope::class)
+class CredentialsSyncMetadata @Inject constructor(
+    private val dao: CredentialsSyncMetadataDao,
+) {
+    fun addOrUpdate(entity: CredentialsSyncMetadataEntity) {
+        dao.insert(entity)
+    }
+
+    fun getSyncId(id: Long): String? {
```

Comment:
it's nice where you have used `syncId` to make it explicit what kind of ID it is; being more explicit that the `id` is a `localId` might also be good in this class

Reply:
also updated at db level, sounds good